pkg/plugin/datasource.go

```diff
@@ -25,13 +29,41 @@ var (
 )
 
 // NewDatasource creates a new datasource instance.
-func NewDatasource(_ backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
-	return &Datasource{}, nil
+func NewDatasource(dataSourceSettings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+
+	log.DefaultLogger.Debug("settings from front end", "json data", dataSourceSettings.JSONData, "secure json data", dataSourceSettings.DecryptedSecureJSONData)
+	var settings map[string]string
+	err := json.Unmarshal(dataSourceSettings.JSONData, &settings)
+	if err != nil {
+		return nil, err
+	}
+	log.DefaultLogger.Debug("parsed json data", settings)
+	bucket := settings["bucket"]
+	id := dataSourceSettings.DecryptedSecureJSONData["aws_access_key_id"]
+	secretKey := dataSourceSettings.DecryptedSecureJSONData["aws_secret_access_key"]
+
+	cfg, err := config.LoadDefaultConfig(
+		context.TODO(),
+		config.WithRegion("us-east-1"),
+		config.WithCredentialsProvider(
+			credentials.NewStaticCredentialsProvider(id, secretKey, ""),
+		),
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	s3PresignClient := s3.NewPresignClient(s3.NewFromConfig(cfg))
```

Comment:
does the s3PresignClient object stay good indefinitely?  Does it need to be refreshed, or should it be created when a request is processed?

Reply:
I think that it does, it only needs to be replaced if you change the config, in which case a new data source instance will be created. The client doesn't store any expiring configuration in memory. how exactly service calls work is spelled out https://aws.github.io/aws-sdk-go-v2/docs/making-requests/#calling-service-operations

you were correct earlier in saying that the signing client doesn't need to make any service calls to AWS to generate a signed reqeust. As far as I can tell this presignClient is totally disconnected from AWS, it just uses the signing algorithm